### PR TITLE
WET-379 Main container working example

### PR DESCRIPTION
**Pull request by josephdiab**

### CODE REVIEW

File: common/main-container/maincontainer-article-en.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{%- include variable-core.liquid -%}
+{%- capture page-title -%}
+    {%- if page.titleH1 -%}
+        {{ page.titleH1 }}
+    {%- elsif page.title -%}
+        {{ page.title }}
+    {%- else -%}
+        Page untitled
+    {%- endif -%}
+{%- endcapture -%}
```
Comment (duboisp) [inline common/main-container/maincontainer-article-en.html:21]:
```suggestion
{%- include variable-core.liquid -%}
{%- capture page-title -%}
	{%- if page.titleH1 -%}
		{{ page.titleH1 }}
	{%- elsif page.title -%}
		{{ page.title }}
	{%- else -%}
		Page untitled
	{%- endif -%}
{%- endcapture -%}
```

**Review by duboisp — CHANGES_REQUESTED** [review]

File: common/main-container/maincontainer-article-en.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<article>
+		<!-- heading -->
+		<h3>Article</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-article-en.html:28]: This need to be an h1

File: common/main-container/maincontainer-article-en.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        &lt;article&gt;
+        &lt;!-- heading --&gt;
+
+        &lt;!-- content page --&gt;
+
+        &lt;!-- Page details section [Must be configured as a &lt;footer&gt; element in this situation] --&gt;
+
+        &lt;/article&gt;
+    &lt;/main&gt;
+    </code></pre>
```
Comment (duboisp) [inline common/main-container/maincontainer-article-en.html:58]: Move the code inside the article and replace the h3 by an h2

File: common/main-container/maincontainer-en.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		{%- else -%}
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<!-- heading -->
+	<h3>Lorem Ipsum</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-en.html:27]:
```suggestion
	<h1 id="wb-cont">Lorem Ipsum</h1>
```

File: common/main-container/maincontainer-en.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+		elit. In varius elementum nunc, nec sollicitudin lorem malesuada at. Duis commodo quis enim ac consectetur.
+		Morbi facilisis tincidunt est, quis auctor lectus commodo eu. In pulvinar mi ligula, eu interdum sem lacinia
+		nec. Phasellus tempus mi id massa laoreet, a pellentesque dui venenatis. Maecenas dignissim, lorem pretium
+		facilisis tincidunt, dui massa iaculis turpis, quis convallis justo justo sit amet quam. Donec consectetur
+		eleifend fringilla. Cras eu rhoncus dui. Nam sit amet libero commodo quam sagittis commodo eu eget nisl.
+		Vestibulum vitae consequat velit. Ut blandit metus id justo efficitur, id maximus quam euismod. Pellentesque
+		eros libero, hendrerit non viverra at, euismod eu nunc. Fusce sit amet tincidunt lacus. Proin dictum arcu at leo
+		aliquam placerat. Donec volutpat ornare condimentum.</p>
+
+	<h3>Code</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-en.html:40]:
```suggestion
	<h2>Code</h2>
```

File: common/main-container/maincontainer-fr.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		{%- else -%}
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<!-- heading -->
+	<h3>Lorem Ipsum</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-fr.html:27]:
```suggestion
	<h1 id="wb-cont">Lorem Ipsum</h1>
```

File: common/main-container/maincontainer-fr.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+		elit. In varius elementum nunc, nec sollicitudin lorem malesuada at. Duis commodo quis enim ac consectetur.
+		Morbi facilisis tincidunt est, quis auctor lectus commodo eu. In pulvinar mi ligula, eu interdum sem lacinia
+		nec. Phasellus tempus mi id massa laoreet, a pellentesque dui venenatis. Maecenas dignissim, lorem pretium
+		facilisis tincidunt, dui massa iaculis turpis, quis convallis justo justo sit amet quam. Donec consectetur
+		eleifend fringilla. Cras eu rhoncus dui. Nam sit amet libero commodo quam sagittis commodo eu eget nisl.
+		Vestibulum vitae consequat velit. Ut blandit metus id justo efficitur, id maximus quam euismod. Pellentesque
+		eros libero, hendrerit non viverra at, euismod eu nunc. Fusce sit amet tincidunt lacus. Proin dictum arcu at leo
+		aliquam placerat. Donec volutpat ornare condimentum.</p>
+
+	<h3>Code</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-fr.html:40]:
```suggestion
	<h2>Code</h2>
```

File: common/main-container/maincontainer-article-fr.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        &lt;article&gt;
+        &lt;!-- heading --&gt;
+
+        &lt;!-- contenu de la page --&gt;
+
+        &lt;!-- Section des détails de la page [Doit être configuré comme un élément &lt;footer&gt; dans cette situation] --&gt;
+
+        &lt;/article&gt;
+    &lt;/main&gt;
+    </code></pre>
```
Comment (duboisp) [inline common/main-container/maincontainer-article-fr.html:58]: Move the code inside the article and replace the h3 by an h2

File: common/main-container/maincontainer-article-fr.html [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<article>
+		<!-- heading -->
+		<h3>Article</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-article-fr.html:28]:
```suggestion
		<h1 id="wb-cont">Article</h1>
```

File: common/main-container/maincontainer-section-en.html [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<section>
+		<!-- heading -->
+		<h3>Section</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-section-en.html:28]:
```suggestion
		<h1 id="wb-cont">Section</h1>
```

File: common/main-container/maincontainer-section-en.html [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        &lt;section&gt;
+        &lt;!-- heading --&gt;
+
+        &lt;!-- content page --&gt;
+
+        &lt;!-- Page details section [Must be configured as a &lt;footer&gt; element in this situation] --&gt;
+
+        &lt;/section&gt;
+        &lt;/main&gt;
+    </code></pre>
```
Comment (duboisp) [inline common/main-container/maincontainer-section-en.html:58]: Move inside the section, the h3 need to be an h2

File: common/main-container/maincontainer-section-fr.html [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+			Page untitled
+		{%- endif -%}
+{%- endcapture -%}
+
+<div class="wb-prettify all-pre hide"></div>
+
+<main class="container" property="mainContentOfPage" resource="#wb-main" typeof="WebPageElement">
+	<section>
+		<!-- heading -->
+		<h3>Section</h3>
```
Comment (duboisp) [inline common/main-container/maincontainer-section-fr.html:28]:
```suggestion
		<h1 id="wb-cont">Section</h1>
```

File: common/main-container/maincontainer-section-fr.html [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+        &lt;section&gt;
+        &lt;!-- heading --&gt;
+
+        &lt;!-- contenu de la page --&gt;
+
+        &lt;!-- Section des détails de la page [Doit être configuré comme un élément &lt;footer&gt; dans cette situation] --&gt;
+
+        &lt;/section&gt;
+    &lt;/main&gt;
+    </code></pre>
```
Comment (duboisp) [inline common/main-container/maincontainer-section-fr.html:57]: Move inside the section and the h3 need to be h2

File: common/main-container/reports/pre-a11y-1.json [hunk 14]
```diff
@@ -0,0 +1,596 @@
...
+    "@id": "_:subject",
+    "dct:references": "https://example.com/web/page/url.html",
+    "@type": [
+      "earl:TestSubject",
+      "schema:WebPage"
+    ],
+    "dct:description": "Pre assessment for the main container component.",
+    "earl:pointer": {
+      "@type": "oa:CssSelector",
+      "@value": "title, meta, #wb-bc, main"
```
Comment (duboisp) [inline common/main-container/reports/pre-a11y-1.json:17]: Here we are only testing the main element, so the selector should be `main`

```suggestion
      "@value": "main"
```

File: package-lock.json
Comment (duboisp) [inline package-lock.json:1]: This file need to be removed from the commit.

**Review by duboisp — CHANGES_REQUESTED** [review]

**Comment by duboisp** [comment]
@AtlakM can you apply the requested changes. Thanks and comment in which of your GitHub branch we can find the changes

**Review by duboisp — CHANGES_REQUESTED** [review]
Tested and reviewed, see the requested changes.

File: common/main-container/index.json-ld [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+				"title": "Main Container with section",
+				"language": "en",
+				"path": "maincontainer-section-en.html"
+			},
+			{
+				"title": "Conteneur Principal avec section",
+				"language": "fr",
+				"path": "maincontainer-section-fr.html"
+			}
+		]
```
Comment (duboisp) [inline common/main-container/index.json-ld:53]: Add the report section like:

Please replace the title and the link to match the reports pre-assessement file on this PR.

```suggestion
		],
		"reports": [
			{
				"title": "Accessibility assessment #1",
				"language": "en",
				"path": "reports/a11y-2-en.html"
			},
			{
				"title": "Évaluation de l'accessibilité #1",
				"language": "fr",
				"path": "reports/a11y-2-fr.html"
			}
		]
```

**Review by duboisp — CHANGES_REQUESTED** [review]

**Comment by AtlakM** [comment]
> @AtlakM can you apply the requested changes. Thanks and comment in which of your GitHub branch we can find the changes

https://github.com/AtlakM/GCWeb/commits/main-container/

File: common/main-container/maincontainer-article-fr.html [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+		&lt;main property="mainContentPage"&gt;
+			&lt;article&gt;
+			&lt;!-- heading --&gt;
+
+			&lt;!-- contenu de la page --&gt;
+
+			&lt;!-- Section des détails de la page [Doit être configuré comme un élément &lt;footer&gt; dans cette situation] --&gt;
+
+			&lt;/article&gt;
+		&lt;/main&gt;
```
Comment (duboisp) [inline common/main-container/maincontainer-article-fr.html:56]: This code sample don't match this page example
```suggestion
		&lt;main property="mainContentPage"&gt;
			&lt;article&gt;
			&lt;!-- heading --&gt;
		&lt;h1&gt;Article&lt;/h1&gt;

			&lt;!-- contenu de la page --&gt;

			&lt;!-- Section des détails de la page [Doit être configuré comme un élément &lt;footer&gt; dans cette situation] --&gt;

			&lt;/article&gt;
		&lt;/main&gt;
```

File: common/main-container/maincontainer-fr.html [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+	<pre><code>
+	&lt;h2&gt;Conteneur Principal&lt;/h2&gt;
+	&lt;main property="mainContentPage"&gt;
+
+		&lt;!-- heading --&gt;
+
+		&lt;!-- contSenu de la page --&gt;
+
+	&lt;/main&gt;
+	</code></pre>
```
Comment (duboisp) [inline common/main-container/maincontainer-fr.html:50]: Code sample don't match

**Review by duboisp — CHANGES_REQUESTED** [review]
A few last change requested. I think this are going to the last change.

**Comment by AtlakM** [comment]
> A few last change requested. I think this are going to the last change.

https://github.com/AtlakM/GCWeb/commit/4be26cc4e65da2504d4ad9e0802ef65d3654558d

**Review by duboisp — APPROVED** [review]
Review and fixed the merge conflict.

This is a content change.

**Review by duboisp — APPROVED** [review]
The JSON parsing error is not fixed. This can be merged on the CI build is completed

**Pull request merged**